audio/Makefile

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (C) 2009-2012 Chris McClelland
+# Copyright 2015 Joel Stanley <joel@jms.id.au>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# To build the firmware you will need:
+#   SDCC from http://sdcc.sourceforge.net
+#   fx2lib from http://fx2lib.wiki.sourceforge.net
+#
+# To build a firmware suitable for loading into RAM:
+#   make
+#
+# To build a firmware suitable for loading from EEPROM:
+#   make FLAGS="-DEEPROM"
+#
+#
+# This Makefile assumes a recent version of SDCC. If you're still using an older
+# version with the asx8051 assembler, build like this:
+#   make AS8051=asx8051
+#
+TARGET := audio
+FX2LIBDIR := ../third_party/fx2lib
+LINUXHEADERSDIR := ../third_party/fx2lib-linux-headers
+LIBS := $(FX2LIBDIR)/lib/fx2.lib
+INCS := -I sdcc -I$(FX2LIBDIR)/include -I$(LINUXHEADERSDIR) -I.
+
+# Settings specific for the TimVideo hdmi2usb firmware
+BOARD ?= opsis
+FLAGS +=-DDEBUG -DBOARD_${BOARD}
+USE_16K := 1
+
+# Must only be hex numbers
+FIRMWARE_VERSION := $(shell date +%Y%m%d)
+
+CC_SRCS := descriptors.c firmware.c
+CC_OBJS := $(CC_SRCS:%.c=%.rel)
+
+CC := sdcc
+
+ifeq (${BOARD},atlys)
+VID := 1D50
+PID := 60B7
+DID := 0002
+else
+ifeq (${BOARD},opsis)
+VID := 2A19
+PID := 5442
+DID := 0002
+else
+$(error "Unknown board type '$(BOARD)'")
```

Comment:
Well, I think at the moment you only have confidence that this works on the fx2dev board, so it might be better to start with that here?